src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/WebClientHelpers.kt

```diff
@@ -1,16 +1,34 @@
 package uk.gov.justice.digital.hmpps.prisonertonomisupdate.helpers
 
+import kotlinx.coroutines.reactor.awaitSingle
 import kotlinx.coroutines.reactor.awaitSingleOrNull
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.WebClientResponseException
 import org.springframework.web.reactive.function.client.bodyToMono
 import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.ErrorResponse
 
 suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrNotFound(): T? =
   this.bodyToMono<T>()
     .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
     .awaitSingleOrNull()
 
+suspend inline fun <reified T : Any> WebClient.ResponseSpec.awaitBodyOrAttendancePaidException(): T =
+  this.bodyToMono<T>()
+    .onErrorResume(WebClientResponseException.BadRequest::class.java) {
+      val errorResponse = it.getResponseBodyAs(ErrorResponse::class.java)
+      if (errorResponse.errorCode == BadRequestError.ATTENDANCE_PAID.errorCode) {
+        Mono.error(AttendancePaidException(errorResponse))
+      } else {
+        Mono.error(it)
+      }
+    }
+    .awaitSingle()
+
+enum class BadRequestError(val errorCode: Int) {
+  ATTENDANCE_PAID(1001),
+}
```

Comment:
I couldn't find a nice way to include the error enum in the Open API spec. so have duplicated here.